tiledb/sm/c_api/tiledb.cc

```diff
@@ -1,3 +1,4 @@
+#define FORCE_QUERY_EXTERNAL_SUBARRAY_USE 0
```

Comment:
Is this for testing? Do we plan to enable this by default/remove the protection and use the new code? Asking in the consideration of the unit test questions to see the value in testing the previous code paths.

Reply:
Would like to leave in place and ship active[*], 
'in place' to have by default only one code path that needs to be considered in general,
but optional old paths ,as insurance against possible failures in exposed newer and newly exposed API implementations to provide a fallback position as quickly as possible if differences/errors are discovered in new and newly exposed code paths that cannot otherwise be addressed rapidly enough.
(My position on this is not shared by all.)
([*]If it were more palatable to all, these compilation optional paths could become runtime configured optional paths, possibly providing an even quicker problem resolution if problems/differences surface in the field.
If runtime configurability were provided, then consideration should be given as to whether to provide finer granularity for enabling the optional paths individually or perhaps per 'group' rather than in their entirety.)

Reply:
Thank you for the additional details here. We should not have `FORCE_QUERY_EXTERNAL_SUBARRAY_USE` in the final version here, lets either upgrade the deprecated functions to the new internal API use or leave them as is. Supporting `FORCE_QUERY_EXTERNAL_SUBARRAY_USE` will add a layer of difficulty in maintaining the API and supporting end users where the behavior might be changed based on compile time options.

I'd be in favor of upgrading the deprecated APIs to reduce the paths we have to support now, and reducing the risk of regression by allowing all the various APIs and integrations to test the new subarray object through the existing `add_range` functionality. I think this also reduces the code surface and we can remove some of the integration tests that would be overlapping based on that many integration tests are really attempting to test core internal logic and no c/cpp-api specific items.